providers/dns/nifcloud/client.go

```diff
@@ -0,0 +1,202 @@
+	} `xml:"Error"`
+	RequestID string `xml:"RequestId"`
+}
+
+// ChangeBatch is the information for a change request.
+type ChangeBatch struct {
+	Changes Changes `xml:"Changes"`
+	Comment string  `xml:"Comment"`
+}
+
+// Changes is array of Change.
+type Changes struct {
+	Change []Change `xml:"Change"`
+}
+
+// Change is the information for each resource record set that you want to change.
+type Change struct {
+	Action            string            `xml:"Action"`
+	ResourceRecordSet ResourceRecordSet `xml:"ResourceRecordSet"`
+}
+
+// ResourceRecordSet is the information about the resource record set to create or delete.
+type ResourceRecordSet struct {
+	Name            string          `xml:"Name"`
+	Type            string          `xml:"Type"`
+	TTL             int             `xml:"TTL"`
+	ResourceRecords ResourceRecords `xml:"ResourceRecords"`
+}
+
+// ResourceRecords is array of ResourceRecord.
+type ResourceRecords struct {
+	ResourceRecord []ResourceRecord `xml:"ResourceRecord"`
+}
+
+// ResourceRecord is the information specific to the resource record.
+type ResourceRecord struct {
+	Value string `xml:"Value"`
+}
+
+// ChangeInfo is A complex type that describes change information about changes made to your hosted zone.
+type ChangeInfo struct {
+	ID          string `xml:"Id"`
+	Status      string `xml:"Status"`
+	SubmittedAt string `xml:"SubmittedAt"`
+}
+
+// NifcloudDNSClient client of NIFCLOUD DNS
+type NifcloudDNSClient struct {
+	accessKey string
+	secretKey string
+}
+
+// NewDNSClient creates NifcloudDNSClient
+func NewDNSClient() *NifcloudDNSClient {
+	return &NifcloudDNSClient{
+		accessKey: os.Getenv("NIFCLOUD_ACCESS_KEY_ID"),
+		secretKey: os.Getenv("NIFCLOUD_SECRET_ACCESS_KEY"),
+	}
+}
+
+// ChangeResourceRecordSets Call ChangeResourceRecordSets API and return response.
+func (c *NifcloudDNSClient) ChangeResourceRecordSets(hostedZoneID string, input ChangeResourceRecordSetsRequest) (output *ChangeResourceRecordSetsResponse, err error) {
+	requestURL := fmt.Sprintf("%s/%s/hostedzone/%s/rrset", apiEndpoint, apiVersion, hostedZoneID)
+
+	body := &bytes.Buffer{}
+	body.Write([]byte(xml.Header))
+	enc := xml.NewEncoder(body)
+	err = enc.Encode(input)
+
+	if err != nil {
+		return
+	}
+
+	req, err := http.NewRequest("POST", requestURL, body)
+
+	if err != nil {
+		return
+	}
+
+	req.Header.Add("Content-Type", "text/xml; charset=utf-8")
+
+	c.sign(req)
+
+	output = new(ChangeResourceRecordSetsResponse)
+
+	res, err := http.DefaultClient.Do(req)
+
+	if err != nil {
+		return
+	}
+
+	if res.StatusCode != 200 {
+		var errResp = ErrorResponse{}
+		err = xml.NewDecoder(res.Body).Decode(&errResp)
+		if err != nil {
+			return
+		}
+		return nil, fmt.Errorf(errResp.Error.Message)
+	}
+
+	err = xml.NewDecoder(res.Body).Decode(&output)
+	defer res.Body.Close()
+	return
+}
+
+// GetChange Call GetChange API and return response.
+func (c *NifcloudDNSClient) GetChange(statusID string) (output *GetChangeResponse, err error) {
```

Comment:
could you use explicit `return output, err` instead of named outputs `(output *GetChangeResponse, err error)`